tests: net: socket: offload_dispatcher move to new ztest API

hunterhu3000 opened the pull request:
Move offload_dispatcher tests to use new ztest API

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

yperess commented on tests/net/socket/offload_dispatcher/src/main.c:
```diff
@@ -436,27 +436,29 @@ static void test_result_reset(void)
 }
 
-static void test_socket_teardown(void)
+static void test_socket_teardown(void *dummy)
```
nit: Add `ARG_UNUSED(dummy)`

hunterhu3000 replied:
OK

yperess commented on tests/net/socket/offload_dispatcher/src/main.c:
```diff
@@ -842,65 +844,39 @@ static void test_tls_native_iface_native(void)
+	    NULL, NULL, test_socket_teardown);
+ZTEST_SUITE(socket_offload_dispatcher_shutdown_test_suite, NULL, test_socket_setup_udp,
+	    NULL, NULL, test_socket_teardown);
+ZTEST_SUITE(socket_offload_bind_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_connect_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_listen_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_accept_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_sendto_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_recvfrom_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_getsockopt_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_setsockopt_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_sendmsg_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_getpeername_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_getsockname_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_so_offload_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_so_native_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
```
These all have the exact same setup/teardown, why can't we just use 1 suite?

hunterhu3000 replied:
only can combine to 3 suite

yperess commented on tests/net/socket/offload_dispatcher/src/main.c:
```diff
@@ -842,65 +844,39 @@ static void test_tls_native_iface_native(void)
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_so_native_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_tls_offload_test_suite, NULL, test_socket_setup_tls, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_tls_native_test_suite, NULL, test_socket_setup_tls, NULL, NULL,
+	    test_socket_teardown);
```
Same here, these are the same

hunterhu3000 replied:
> Same here, these are the same

The name looks similar
socket_offload_tls_offload_test_suite contains the case of test_tls_native_iface_offloaded
socket_offload_tls_native_test_suite contains the test of test_tls_native_iface_native
They are different

hunterhu3000 replied:
In my code setup / teardown will be run before / after each test.
If combine the test cases, setup / teardown will be run only once for each test suite

yperess reviewed (changes requested):
There seems to be a lot of unnecessary test suites generated for this test

hunterhu3000 committed:
tests: net: socket: offload_dispatcher move to new ztest API

Move offload_dispatcher tests to use new ztest API
TEST=twister -T tests/net/socket/offload_dispatcher

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

hunterhu3000 commented:
In this test case CONFIG_ZTEST_SHFFLE=y can be turned on.
Thanks to review.

hunterhu3000 commented:
@yperess I have modified per your request and reduced the test case number to 3. If it is OK to you, could you please review and +1? So that the patch can be merged into the mainline. Thanks
